crates/uv/tests/it/pip_compile.rs

```diff
@@ -15230,6 +15230,274 @@ fn respect_index_preference() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn dependency_group() -> Result<()> {
+    fn new_context() -> Result<TestContext> {
+        let context = TestContext::new("3.12");
+
+        let pyproject_toml = context.temp_dir.child("pyproject.toml");
+        pyproject_toml.write_str(
+            r#"
+            [project]
+            name = "project"
+            version = "0.1.0"
+            requires-python = ">=3.12"
+            dependencies = ["typing-extensions"]
+            [dependency-groups]
+            foo = ["sortedcontainers"]
+            bar = ["iniconfig"]
+            dev = ["sniffio"]
+            "#,
+        )?;
+
+        Ok(context)
+    }
+
+    let mut context;
+
+    context = new_context()?;
+    uv_snapshot!(context.filters(), context.pip_compile()
+        .arg("--group").arg("bar"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --group bar
+    iniconfig==2.0.0
+
+    ----- stderr -----
+    Resolved 1 package in [TIME]
+    ");
+
+    context = new_context()?;
+    uv_snapshot!(context.filters(), context.pip_compile()
+        .arg("pyproject.toml")
```

Comment:
It might be nice to have a comment before these test cases about what you're covering? Like

> // Compile the project dependencies and a dependency group